### PR TITLE
add deployment story for lecture series infra

### PR DESCRIPTION
**Pull request by ctem**
This is one SoN participant’s humble retelling of the legend of the 2022 infra project—the birth of an era.

<del>
Note: while the document itself is ready for review, the way the site renders it is a little wonky at this point:

- Markdown isn’t supported in titles
- inline code blocks and tables render with no backgrounds or borders
- cramped whitespace between sections presents a risk to structural clarity
</del>

(Rendering issues addressed in [#39](https://github.com/NixOS/nixos-summer/pull/39).)

### CODE REVIEW

File: content/blog/the-rise-of-special-project-infra.md [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+
+Everybody needs a saga. Ours was: self-host a secure livestreaming server to accommodate a sizable audience participating from all around the world and at variable network speeds, _fast_. Oh, and add a chat room for synchronous Q&A.
+
+On the server side, we settled on the following stack: virtual private server (VPS) → NixOS → [Caddy](https://caddyserver.com/) → [Owncast](https://owncast.online/).
+
+The media would originate on our intrepid moderator’s local [OBS Studio](https://obsproject.com/) and be streamed at archival quality over the Real-Time Messaging Protocol (RTMP) to our remote Owncast service. Owncast would transcode the video to multiple stream qualities (i.e., optimized variants of video bitrate and CPU usage) and present the selected variant to the user in an attractive, branded web UI with a convenient chat box.
+
+### Phase II: Provision
+
+Controlling the narrative has always called for a bit of hardware. For this exercise, we went with a single VPS equipped with dedicated vCPUs. The general provisioning procedure follows:
```
Comment (fricklerhandwerk) [inline content/blog/the-rise-of-special-project-infra.md:32]: > Controlling the narrative has always called for a bit of hardware.

I'm not sure what that sentence is supposed to mean. Which narrative?

File: content/blog/the-rise-of-special-project-infra.md [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+
+1. Choose a (preferably reputable and carbon negative) VPS provider.
+
+1. Provision an instance appropriate to the task and estimated audience.
+
+   - To minimize latency, select the available region that best approximates the location of the source stream as well as the mean location of the majority of expected participants where applicable. (Note: _object storage_ and _CDN caching_ additionally may be leveraged to accommodate participants; object storage enables the streaming service to serve media files when the concurrent user count exceeds available bandwidth, and CDN caching enables users to download these assets from the nearest available server.)
+   - Select a model with enough (preferably dedicated) vCPU cores to handle video transcoding, which is a highly processor-intensive operation. (Note: this should correspond to the stream variants selected in the [streaming service configuration](#streaming-service-configuration).)
```
Comment (fricklerhandwerk) [inline content/blog/the-rise-of-special-project-infra.md:39]:
```suggestion
   - Select a server model with enough (preferably dedicated) vCPU cores to handle video transcoding, which is a highly processor-intensive operation. (Note: this should correspond to the stream variants selected in the [streaming service configuration](#streaming-service-configuration).)
```

File: content/blog/the-rise-of-special-project-infra.md [hunk 3]
```diff
@@ -0,0 +1,168 @@
...
+        reverse_proxy ${owncastWebService}
+      '';
+    };
+  };
+}
+```
+
+Demonstrating the great UX of a well-written NixOS module, the Owncast configuration is completely out-of-the-box; using [all defaults](https://search.nixos.org/options?channel=unstable&from=0&size=50&sort=relevance&type=options&query=owncast), one line installs and activates the service, initializes all necessary state (including a SQLite database `owncast.db` that stores important service configuration such as administrative credentials) in `/var/lib/owncast`, and binds the web server to `127.0.0.1:8080`.
+
+The Caddy module is similarly loaded (with goodies). The only customization necessary here was to specify the domain we prepared (in the last step of the provisioning phase above) as a virtual host from which Caddy can forward client requests to the backend Owncast web server. For good measure, we also specified an email address for Caddy to use when setting up SSL on our behalf. (This is the email address to use for account creation and correspondence from the SSL certificate authority, which is [Let's Encrypt](https://letsencrypt.org/) by default.)
```
Comment (fricklerhandwerk) [inline content/blog/the-rise-of-special-project-infra.md:83]:
```suggestion
The Caddy module is similarly loaded (with goodies). The only customization necessary here was to specify the domain we prepared (in the last step of the provisioning phase above) as a virtual host from which Caddy can forward client requests to the backend Owncast web server. For good measure, we also specified an email address for Caddy to use when setting up SSL on our behalf. This is the email address to use for account creation and correspondence from the SSL certificate authority, which is [Let's Encrypt](https://letsencrypt.org/) by default.
```

File: content/blog/the-rise-of-special-project-infra.md [hunk 4]
```diff
@@ -0,0 +1,168 @@
...
+
+The remaining configuration simply opens ports in the system firewall for HTTP, HTTPS, and RTMP. (Note that the RTMP port is also an Owncast module-specified default.)
```
Comment (fricklerhandwerk) [inline content/blog/the-rise-of-special-project-infra.md:85]:
```suggestion
The remaining configuration simply opens ports in the system firewall for HTTP, HTTPS, and RTMP. Note that the RTMP port is also an Owncast module-specified default.
```

File: content/blog/the-rise-of-special-project-infra.md [hunk 5]
```diff
@@ -0,0 +1,168 @@
...
+
+That’s the entirety of the NixOS configuration specific to our use case; all additional configuration (e.g., importing generated hardware configuration, enabling SSH and adding appropriate keys, defining Nix garbage collection rules, etc.) is simply boilerplate.
+
+### Phase IV: Deploy
+
+With the instance provisioned and the configuration prepared, it was about time to introduce the two, marking the beautiful beginning of a successful working relationship. This is generally a two-step process:
+
+1. Install NixOS on the instance. Note: a provider may make this very easy (by providing a template image), moderately easy (by allowing [a custom ISO](https://github.com/nix-community/nixos-generators) to be uploaded), or hit-or-miss (by providing an image for a non-NixOS distribution from which NixOS may be converted with [NixOS-Infect](https://github.com/elitak/nixos-infect/) or [`NIXOS_LUSTRATE`](https://nixos.org/manual/nixos/stable/#sec-installing-from-other-distro)).
+
+   Our provider not doing us too many favors but at least known to play nice with `nixos-infect`ed installations, we went with that.
```
Comment (fricklerhandwerk) [inline content/blog/the-rise-of-special-project-infra.md:95]:
```suggestion
   Our provider was not doing us too many favors, but as it is at least known to play nice with `nixos-infect`ed installations, we went with that.
```

I suppose you meant that?

File: content/blog/the-rise-of-special-project-infra.md [hunk 6]
```diff
@@ -0,0 +1,168 @@
...
+
+As previously mentioned, much of the Owncast configuration is stored as mutable state, enabling the service to be modified on-the-fly from the administrative web UI. A suggested procedure follows:
+
+1.  Log in to the administrative web UI:
+
+    - URL: `https://<target FQDN>/admin/` (e.g., `https://live.nixos.org/admin/`)
+    - User name: `admin`
+    - Password: `<stream key>` (`abc123` by default)
+
+1.  In server settings, set a new Stream Key. (Note that this is also the administrative web UI password.)
```
Comment (fricklerhandwerk) [inline content/blog/the-rise-of-special-project-infra.md:114]:
```suggestion
1.  In server settings, set a new Stream Key.

    Note that this is also the administrative web UI password.
```

File: content/blog/the-rise-of-special-project-infra.md [hunk 7]
```diff
@@ -0,0 +1,168 @@
...
+
+1.  In general settings, modify instance details, tags, and social handles as appropriate.
+
+1.  In video settings:
+
+    - Add stream variants to enable adaptive bitrate streaming to accommodate users on various-quality networks. The following table may be used as a guideline:
```
Comment (fricklerhandwerk) [inline content/blog/the-rise-of-special-project-infra.md:120]:
```suggestion
    - Add stream variants to enable adaptive bitrate streaming to accommodate users on various-quality networks.
    
      The following table may be used as a guideline:
```

File: content/blog/the-rise-of-special-project-infra.md [hunk 8]
```diff
@@ -0,0 +1,168 @@
...
+      | x264    | FHD  | 1920x1080        | 1900-4500 kbps | 25 or 30 fps |
+
+      Note: some services recommend a higher bitrate for HD:
+
+      | Encoder | Name | Resolution (WxH) |   Bitrate | Framerate    |
+      | ------- | ---- | ---------------- | --------: | ------------ |
+      | x264    | HD   | 1280x720         | 3000 kbps | 25 or 30 fps |
+      | x264    | FHD  | 1920x1080        | 4500 kbps | 25 or 30 fps |
+
+      Note: our stress tests showed that the use of standard variants with our VPS were reportedly adequate for most users but problematic for some in Japan, Thailand, and the UK. Based on these results, object storage and CDN caching are recommended.
```
Comment (fricklerhandwerk) [inline content/blog/the-rise-of-special-project-infra.md:135]:
```suggestion
      Our stress tests showed that the use of standard variants with our VPS were reportedly adequate for most users but problematic for some in Japan, Thailand, and the UK. Based on these results, object storage and CDN caching are recommended.
```

File: content/blog/the-rise-of-special-project-infra.md [hunk 9]
```diff
@@ -0,0 +1,168 @@
...
+      Additional notes:
+
+      - For interactive live streams (e.g., lectures with Q&A sessions), consider decreasing the latency buffer to Low.
+      - For further tuning, consider referring to a bitrate calculator (e.g., [Bitrate Calc](https://bitratecalc.com/)).
+
+#### Streaming client configuration
+
+With the server side all set, it was time to get streaming! The basic procedure follows:
+
+1. On a local workstation, install and launch [OBS Studio](https://obsproject.com/) ([obs-studio](https://search.nixos.org/packages?channel=22.05&show=obs-studio&from=0&size=50&sort=relevance&type=options&query=obs-studio) in Nixpkgs).
```
Comment (fricklerhandwerk) [inline content/blog/the-rise-of-special-project-infra.md:146]:
```suggestion
1. On a local workstation, install and launch [OBS Studio](https://obsproject.com/) ([obs-studio](https://search.nixos.org/packages?channel=22.05&show=obs-studio&from=0&size=50&sort=relevance&type=options&query=obs-studio) in `nixpkgs`).
```

File: content/blog/the-rise-of-special-project-infra.md [hunk 10]
```diff
@@ -0,0 +1,168 @@
...
+   | ---: | ----: | ------------: | -------- |
+   |    8 | 32 GB |        240 GB | Germany  |
+
+1. Reserve a public IPv4 address and IPv6 subnet and associate with the provisioned instance.
+
+1. Assuming registration of the desired domain name, use a/the DNS registrar-provided interface to update the (or create an) A record for the target subdomain (`live.nixos.org` in our case) with the IP address associated with the provisioned instance. (Note that propagation may take up to 48 hours.)
+
+### Phase III: Configure
+
+Perhaps unintuitively, this was the easiest part. Thanks to the Nixpkgs community contributions of high-quality modules for great modern services, configuring a fully self-hosted livestreaming service safely behind a reverse proxy was scarcely more involved than toggling `enable`. Consider the following Nix code:
```
Comment (fricklerhandwerk) [inline content/blog/the-rise-of-special-project-infra.md:53]:
```suggestion
Perhaps unintuitively, this was the easiest part. Thanks to the [`nixpkgs`](https://github.com/NixOS/nixpkgs) community contributions of high-quality modules for great modern services, configuring a fully self-hosted livestreaming service safely behind a reverse proxy was scarcely more involved than toggling `enable`. Consider the following Nix code:
```

File: content/blog/the-rise-of-special-project-infra.md [hunk 11]
```diff
@@ -0,0 +1,168 @@
...
+
+1. Go live.
+
+1. Confirm the stream at the designated URL (`https://live.nixos.org` in our case).
+
+_It worked?_ It worked. Crisis averted with time to spare. 🍹
+
+## Next Steps
+
+Following a successful first lecture, the SoN infra team is concerned not only with maintenance and optimization of the infrastructure we’ve already deployed but also with innovation on a larger scale. The ultimate goal is to contribute a fully self-hosted, documented, NixOS-based, general-purpose, FLOSS[^floss] conference suite that not only is usable for the SoN Public Lecture Series but also can scale for a modern conference experience—namely [NixCon](https://nixcon.org/).
```
Comment (fricklerhandwerk) [inline content/blog/the-rise-of-special-project-infra.md:162]:
```suggestion
Following a successful first lecture, the SoN infra team is concerned not only with maintenance and optimization of the infrastructure we’ve already deployed but also with innovation on a larger scale. The ultimate goal is to contribute a fully self-hosted, documented, NixOS-based, general-purpose, FLOSS[^floss] conference suite that not only is usable for the SoN Public Lecture Series, but also can scale for a modern conference experience—namely [NixCon](https://nixcon.org/).
```

File: content/blog/the-rise-of-special-project-infra.md [hunk 12]
```diff
@@ -0,0 +1,168 @@
...
+
+In addition to main conference components (e.g., administrative tools, breakout rooms, etc.) and general convenience features (e.g., collaboration tools such as shared whiteboards), a notable priority is the improvement of our accessibility story, e.g., through integration of the [Vosk speech recognition toolkit](https://alphacephei.com/vosk/) for livestream closed captioning/transcription/subtitling (in multiple languages).
+
+To this end, we’re working together with other great teams such as the SoN Special Project for [Jitsi](https://jitsi.org/) and have received guidance and insight from the gracious maintainers of the [NixCon 2020 livestreaming infrastructure](https://github.com/nixcon/nixcon-video-infra)—the crucial project of which our efforts will be a continuation.
```
Comment (fricklerhandwerk) [inline content/blog/the-rise-of-special-project-infra.md:166]:
```suggestion
To this end, we’re working together with other teams such as the SoN Special Project for [Jitsi](https://jitsi.org/) and have received guidance and insight from the gracious maintainers of the [NixCon 2020 livestreaming infrastructure](https://github.com/nixcon/nixcon-video-infra)—the crucial project of which our efforts will be a continuation.
```

I try to abstain from value judgements for the sake of clarity, unless you would want to explain what makes anything "great".

**Review by fricklerhandwerk — APPROVED** [review]
Really good write-up. I like the precise style of explanations a lot. Left a bunch of comments on cosmetics.

There is some general critique on the writing style:
- There are many side notes that break the reading flow. Consider including them in the sentences (I added some suggestions where it appeared obvious) or moving them to footnotes, if they really are not that important.
- Your sentences are often very long. I know this is how your train of thought goes, and it all makes sense. Yet, it's a cognitive burden on the reader who has to build up the stack to understand where you're going. Consider breaking up sentences now and then, and refactor them to avoid nesting.
- This is somewhat personal to me, but not using spaces around emdash (—) makes it really hard to discern insertions from word compositions. I always read "Project—tidily" and "infra—was", and that is just confusing. I know this is how long a go someone with a pointy hat decided to typeset American English, and it just makes me sad. It's such an arbitrary and counterproductive rule. In German you would use endash (–) with spaces around. I don't care which, we're on the internet and not in a 19th century book store, but please separate words with spaces. In general this still goes into the above point - consider using nested sentences sparingly, as they tend to break the reading flow and require the reader to build up stacks. Fragments are more suitable to be put in parentheses, but the same rule of frugality should apply.

File: content/blog/the-rise-of-special-project-infra.md (hunk 2 above)
Comment (ctem) [reply content/blog/the-rise-of-special-project-infra.md:39]: Sure, let’s just go with “server”!

File: content/blog/the-rise-of-special-project-infra.md (hunk 3 above)
Comment (ctem) [reply content/blog/the-rise-of-special-project-infra.md:83]: 👍

File: content/blog/the-rise-of-special-project-infra.md (hunk 6 above)
Comment (ctem) [reply content/blog/the-rise-of-special-project-infra.md:114]: 👍

File: content/blog/the-rise-of-special-project-infra.md (hunk 7 above)
Comment (ctem) [reply content/blog/the-rise-of-special-project-infra.md:120]: 👍

File: content/blog/the-rise-of-special-project-infra.md (hunk 8 above)
Comment (ctem) [reply content/blog/the-rise-of-special-project-infra.md:135]: 👍

File: content/blog/the-rise-of-special-project-infra.md (hunk 9 above)
Comment (ctem) [reply content/blog/the-rise-of-special-project-infra.md:146]: I’m curious about the reasoning behind this one. Throughout our official documentation (including the Nixpkgs manual itself), “Nixpkgs” is generally written as a proper noun unless specifying, e.g., an arbitrary name for a channel or flake input, which is not the case here.

File: content/blog/the-rise-of-special-project-infra.md (hunk 12 above)
Comment (ctem) [reply content/blog/the-rise-of-special-project-infra.md:166]: While I wholeheartedly agree in the context of technical writing, this blog post is in its current form more of an opinion piece styled to spark interest in SoN, the Nix Ecosystem, and self-hosting with Free Software – and as such is brimming with value judgements. Granted the technical density of the instructional content caters to the target readership, documentation-grade objectivity wasn’t a general objective.

Ultimately, I’m trying to understand the scope of this suggestion. I certainly don’t mind taking out simple modifiers like “great”, but unless I’m reading too far into the suggestion, the removal of embellishments like this could be a slippery slope.

**Comment by ctem** [comment]
@fricklerhandwerk Thank you so much for your excellent feedback. Your assistance here and with the Nix Ecosystem documentation is very confidence-inspiring.

I’ll take another pass to address your notes, including more seamless integration of side notes and breaking up longer sentences.

Regarding dashes: while em dashes without surrounding spaces for the use case in question are consensus among the academic style guides with which I’m most familiar, I acknowledge that SoN is funded by the European Commission and as such am willing to reconcile the discrepancy by observing the _[English Style Guide](https://ec.europa.eu/info/sites/default/files/styleguide_english_dgt_en.pdf)_’s preference for en dashes surrounded by spaces. Incidentally, this corresponds to your favored punctuation from German, which for this blog I of course would be willing to accommodate even in the absence of an authoritative style guide.

File: content/blog/the-rise-of-special-project-infra.md (hunk 4 above)
Comment (ctem) [reply content/blog/the-rise-of-special-project-infra.md:85]: 👍

File: content/blog/the-rise-of-special-project-infra.md (hunk 5 above)
Comment (ctem) [reply content/blog/the-rise-of-special-project-infra.md:95]: While I don’t consider the original sentence ungrammatical – think `A doing B, C did D.` – I’m happy to restructure it for clarity.

File: content/blog/the-rise-of-special-project-infra.md (hunk 10 above)
Comment (ctem) [reply content/blog/the-rise-of-special-project-infra.md:53]: Would this additionally apply to “high-quality modules”, etc.? (Please see my reply to your comment on “greatness”.)

File: content/blog/the-rise-of-special-project-infra.md (hunk 11 above)
Comment (ctem) [reply content/blog/the-rise-of-special-project-infra.md:162]: In this sentence, “that” serves as the subject of a declarative content clause. My understanding is that introducing a comma here without a subject would violate the parallel structure. While this could be corrected by simply repeating the subject – `...suite that not only is A, but that also can do B.` – the redundancy feels a bit awkward.

In any case, I’m happy to rework the sentence if it feels too long as-is.

File: content/blog/the-rise-of-special-project-infra.md (hunk 1 above)
Comment (ctem) [reply content/blog/the-rise-of-special-project-infra.md:32]: The expression _controlling the narrative_ (or _framing the discourse_, as it’s sometimes phrased) is used here as a set phrase. In the context of self-hosting, this implies maintaining ownership of both content and its presentation, thereby offering users a way to engage with said content free from third-party influence.

As both the expression and the term _hardware_ are open to interpretation, however, identification of subtext is an exercise left for the reader.

Comment (fricklerhandwerk) [reply content/blog/the-rise-of-special-project-infra.md:32]: I almost supposed that, but it's not as self-explanatory as "framing the narrative", which would actually play nicer with the hardware part: framing the narrative with a bit of hardware.

File: content/blog/the-rise-of-special-project-infra.md (hunk 12 above)
Comment (fricklerhandwerk) [reply content/blog/the-rise-of-special-project-infra.md:166]: It's clear from your style that this is an entertainment and opinion piece sprinkled with technical instructions. I'm always fine with mixing in opinion and value judgement in this context if it it's at least minimally qualified. Just being great does not tell much.

File: content/blog/the-rise-of-special-project-infra.md (hunk 10 above)
Comment (fricklerhandwerk) [reply content/blog/the-rise-of-special-project-infra.md:53]: While it's not clear what high quality would entail exactly (I think it would be valuable to readers if you could qualify this really briefly), "great" is just to broad and overloaded to really mean anything at all.

Again this is personal, and really not that important, but I am a bit annoyed every time someone uses "modern", because it also tells us nothing really, except that something is "recent", or currently en vogue for some reason. Being new or popular does not necessarily mean being good for any definition of "good" (except new or popular, that is).

That's why I suggest better not saying nothing at all. :)

File: content/blog/the-rise-of-special-project-infra.md (hunk 5 above)
Comment (fricklerhandwerk) [reply content/blog/the-rise-of-special-project-infra.md:95]: Yeah, it makes sense on second thought, but really hard to parse without punctuation. I'd prefer clarity over elegance every time.

File: content/blog/the-rise-of-special-project-infra.md (hunk 11 above)
Comment (fricklerhandwerk) [reply content/blog/the-rise-of-special-project-infra.md:162]: Yes please, it was hard to figure out. I'm not all that proficient with English grammar, I only care about it being easy to read.

File: content/blog/the-rise-of-special-project-infra.md (hunk 9 above)
Comment (fricklerhandwerk) [reply content/blog/the-rise-of-special-project-infra.md:146]: First of all, do not consider the manuals authoritative, they've just grown randomly over 19 years and are all over the place with regard to writing style and even terminology.

We haven't figured this out yet, and here is the single source of truth where it should happen eventually, hopefully: https://github.com/NixOS/nix.dev/issues/263

My consideration is this: Nixpkgs is "not a word" because you can't properly read it like other words. You have to know it means something else, namely "Nix packages". `nixpkgs` does not have to be a word, it's a code token that describes something build in a computer language, it's meaning is the encoding it refers to. You don't have to "read" it, but there should be a canonical pronunciation which derives from the original meaning: Nix packages.

Since "Nix packages" is awkward to use as a subject ("X is in Nix packages", "Nix packages is a Y", and I think the thing should be singular as it describes a closed concept) I proposed "Nix package collection". The repository is still awkwardly described as "Nix packages collection" and I would like to change that once we fix terminology.

File: content/blog/the-rise-of-special-project-infra.md (hunk 5 above)
Comment (ctem) [reply content/blog/the-rise-of-special-project-infra.md:95]: 👍

File: content/blog/the-rise-of-special-project-infra.md (hunk 9 above)
Comment (ctem) [reply content/blog/the-rise-of-special-project-infra.md:146]: Thanks for the explanation. I’ll go ahead and change all occurrences of “Nixpkgs” to “`nixpkgs`” for this article.

File: content/blog/the-rise-of-special-project-infra.md (hunk 11 above)
Comment (ctem) [reply content/blog/the-rise-of-special-project-infra.md:162]: 👍

File: content/blog/the-rise-of-special-project-infra.md (hunk 12 above)
Comment (ctem) [reply content/blog/the-rise-of-special-project-infra.md:166]: 👍

File: content/blog/the-rise-of-special-project-infra.md (hunk 1 above)
Comment (ctem) [reply content/blog/the-rise-of-special-project-infra.md:32]: 👍

**Review by bryanhonof — APPROVED** [review]

File: content/blog/the-rise-of-special-project-infra.md (hunk 10 above)
Comment (ctem) [reply content/blog/the-rise-of-special-project-infra.md]: Thanks for the explanation. In this case, “great modern services” referred to the ones described in the article. Changing this to “many services” would’ve changed the meaning, so for the latest revision I took the long way around and added footnotes for clarity.

File: content/blog/the-rise-of-special-project-infra.md [hunk 13]
```diff
@@ -0,0 +1,190 @@
+---
+title: The Rise of Special Project _infra_
+date: 2022-08-12
+extra:
+  author: Ctem
+---
+
+_72 hours out. A bead of sweat slides from your brow, falls to the marred chassis of your local build server, and sizzles into mist, leaving a scant salt stain to tell the tale. It’s the start of the hottest Summer of Nix on record, and you have three days to [research](#phase-i-research), [provision](#phase-ii-provision), [configure](#phase-iii-configure), and [deploy](#phase-iv-deploy). With or without you, this lecture is going live._
+
+The deadline was 19 July, 2022; at five o’clock on that glistening Tuesday afternoon, one Eelco Dolstra – a living legend to those who understand – would take to the webcam from the verdant city of Utrecht to deliver a highly anticipated slice of fresh perspective on his now-19-year-old brainchild, [Nix](https://nixos.org/learn.html). It was the inaugural event in the premier Summer of Nix (SoN) Public Lecture Series. The hype was real, but so was our predicament: no self-hosted livestreaming infrastructure was yet in place.
```
Comment (fricklerhandwerk) [inline content/blog/the-rise-of-special-project-infra.md:10]: Is this a typo?
```suggestion
The deadline was 19 July, 2022; at five o’clock on that glistening Tuesday afternoon, one Eelco Dolstra – a living legend to those who understand – would talk to the webcam from the verdant city of Utrecht to deliver a highly anticipated slice of fresh perspective on his now-19-year-old brainchild, [Nix](https://nixos.org/learn.html). It was the inaugural event in the premier Summer of Nix (SoN) Public Lecture Series. The hype was real, but so was our predicament: no self-hosted livestreaming infrastructure was yet in place.
```

**Review by fricklerhandwerk — APPROVED** [review]

**Review by bryanhonof — APPROVED** [review]

File: content/blog/the-rise-of-special-project-infra.md (hunk 13 above)
Comment (ctem) [reply content/blog/the-rise-of-special-project-infra.md:10]: “Take to the [platform/medium]" is an expression, where _platform/medium_ can be practically anything (e.g., stage, podium, mic, pen, quill). It’s a bit of literary flair to complement the energetic tone of the paragraph.

Comment (fricklerhandwerk) [reply content/blog/the-rise-of-special-project-infra.md:10]: Okay, please don't use idioms in technical writing. Most people who may be able to read your article ([1.5 B](https://en.m.wikipedia.org/wiki/List_of_languages_by_total_number_of_speakers#Top_languages_by_population)) don't know enough idiomatic English to make sense of them. (Sample size 1: 20 years of exposure and 10 years of daily English practice is apparently not enough.)

Possibly self-explanatory idioms are fine. See [controlling the narrative vs. framing the discourse](https://github.com/NixOS/nixos-summer/pull/38/files#diff-44d8314b85b5138271eaabb0c64e176694dbff2c03a4e29bd74e3fb90da1b162R36).

**Comment by MMesch** [comment]
@ctem @fricklerhandwerk is this ready?

**Review by bryanhonof — APPROVED** [review]

**Review by fricklerhandwerk — APPROVED** [review]
